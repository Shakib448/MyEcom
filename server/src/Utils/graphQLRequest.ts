import app from "./server";
import supertest from "supertest";

const graphQLRequest = (query: string, token?: string) => {
  return supertest(app)
    .post("/graphql")
    .send({
      query,
    })
    .set("Authorization", `Bearer ${token}`);
};

export default graphQLRequest;
