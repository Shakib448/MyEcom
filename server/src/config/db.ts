import mongoose from "mongoose";
import "colors";

const dbURL = "mongodb://localhost:27017/myEcom";

export async function connectDB(): Promise<any> {
  await mongoose.connect(dbURL);
  console.log("Database connected successfully".cyan.italic);
}

export async function stopDB(): Promise<void> {
  await mongoose.connection.close();
}
