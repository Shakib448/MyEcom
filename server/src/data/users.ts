import bcrypt from "bcryptjs";

const users = [
  {
    firstName: "Admin",
    lastName: "User",
    email: "admin@example.com",
    password: bcrypt.hashSync("123456789", 10),
    phoneNumber: "123456789",
    address: "Mirpur-1",
    city: "Dhaka",
    state: "Dhaka",
    country: "Bangladesh",
    zip: "1216",
    location: "House-9",
    isAdmin: true,
  },
  {
    firstName: "John",
    lastName: "Doe",
    email: "john@example.com",
    password: bcrypt.hashSync("123456789", 10),
    phoneNumber: "123456789",
    address: "Mirpur-1",
    city: "Dhaka",
    state: "Dhaka",
    country: "Bangladesh",
    zip: "1216",
    location: "House-9",
    isAdmin: true,
  },
  {
    firstName: "Muktadir",
    lastName: "Ahamed",
    email: "muktadir@example.com",
    password: bcrypt.hashSync("123456789", 10),
    phoneNumber: "123456789",
    address: "Mirpur-1",
    city: "Dhaka",
    state: "Dhaka",
    country: "Bangladesh",
    zip: "1216",
    location: "House-9",
    isAdmin: true,
  },
];

export default users;
