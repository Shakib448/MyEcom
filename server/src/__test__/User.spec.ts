import { connectDB, stopDB } from "../config/db";
import User from "../models/User.model";
import graphQLRequest from "../Utils/graphQLRequest";
import users from "../data/users.data";

describe("User testing", () => {
  beforeAll(async () => {
    await connectDB();
    await User.deleteMany();
  });
  afterAll(async () => {
    await User.insertMany(users);
    await stopDB();
  });

  it("Module should be define", () => {
    expect(User).toBeDefined();
  });

  it("User will be created with success message", async () => {
    const req = await graphQLRequest(
      `mutation UserCreate {
      userCreate(
        email: "test@test.com"
        password: "12345678"
        confirmPassword: "12345678"
        firstName: "testFirst name"
        lastName: "Test"
        phoneNumber: "1234567"
        address: "test"
        city: "test"
        country: "test"
        state: "test"
        location: "test"
        zip: "12345"
      ) {
        message
        success
        user {
          email
          firstName
          lastName
          phoneNumber
          address
          city
          zip
          location
          state
          country
        }
      }
    }`
    );
    expect(req.body.data.userCreate.message).toBe("User created successfully");
    expect(req.body.data.userCreate.success).toBe(true);
    expect(req.body.data.userCreate.user).toBeInstanceOf(Object);
    expect(req.body.data.userCreate).toMatchSnapshot();
  });

  it("User will be logged in successfully with success message", async () => {
    const req = await graphQLRequest(
      `mutation AuthUser {
      authUser(email : "test@test.com" password : "12345678"){
        message
        success
        user {
          email
          firstName
          lastName
          phoneNumber
          address
          city
          zip
          location
          state
          country
          token
        }
      }
    }`
    );

    expect(req.body.data.authUser.message).toBe("User logged in successfully");
    expect(req.body.data.authUser.success).toBe(true);
    expect(req.body.data.authUser.user.token).toBeDefined();
    expect(req.body.data.authUser.user).toBeInstanceOf(Object);
  });

  it("User will be updated with Bearer token also find by id with success message", async () => {
    const resToken = await graphQLRequest(
      `mutation AuthUser {
      authUser(email : "test@test.com" password : "12345678"){
        message
        success
        user {
          email
          firstName
          lastName
          phoneNumber
          address
          city
          zip
          location
          state
          country
          token
        }
      }
    }`
    );
    const req = await graphQLRequest(
      `mutation UpdateUser {
      updateUser(
        email: "test@test.com"
        password: "update123456"
        confirmPassword: "update123456"
        firstName: "update name"
        lastName: "update"
        phoneNumber: "1234567"
        address: "update address"
        city: "update city"
        country: "update country"
        state: "update state"
        location: "update location"
        zip: "12345"
      ) {
        message
        success
        user {
          email
          firstName
          lastName
          phoneNumber
          address
          city
          zip
          location
          state
          country
        }
      }
    }`,
      resToken.body.data.authUser.user.token
    );

    expect(req.body.data.updateUser.message).toBe("User updated successfully");
    expect(req.body.data.updateUser.success).toBe(true);
    expect(req.body.data.updateUser.user).toBeInstanceOf(Object);
    expect(req.body.data.updateUser).toMatchSnapshot();

    const findById = await graphQLRequest(
      `mutation GetUserProfile {
        getUserProfile(id: "${resToken.body.data.authUser.user.id}") {
        id
        email
        firstName
        lastName
        phoneNumber
        address
        city
        country
        state
        zip
        location
      }
    }`,
      resToken.body.data.authUser.user.token
    );
    expect(findById.body.data.getUserProfile).toBeInstanceOf(Object);
  });

  it("User should be delete with success message", async () => {
    const req = await graphQLRequest(`query GetAllUser {
      getAllUsers {
        id
        email
        firstName
        lastName
        phoneNumber
        address
        city
        country
        state
        zip
        location
      }
    }`);

    const userId = req.body.data.getAllUsers.map((item: any) => item.id);

    const deleteUser = await graphQLRequest(`mutation UserDeleteById{
      userDeleteById(id: "${userId[0]}") {
        success
        message
      }
    }`);
    expect(deleteUser.body.data.userDeleteById).toBeInstanceOf(Object);
    expect(deleteUser.body.data.userDeleteById.success).toBe(true);
    expect(deleteUser.body.data.userDeleteById.message).toBe(
      "User deleted successfully"
    );
    expect(deleteUser.body.data.userDeleteById).toMatchSnapshot();
  });
});
