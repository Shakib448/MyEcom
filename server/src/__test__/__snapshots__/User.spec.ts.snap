// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User testing User should be delete with success message 1`] = `
Object {
  "message": "User deleted successfully",
  "success": true,
}
`;

exports[`User testing User will be created with success message 1`] = `
Object {
  "message": "User created successfully",
  "success": true,
  "user": Object {
    "address": "test",
    "city": "test",
    "country": "test",
    "email": "test@test.com",
    "firstName": "testFirst name",
    "lastName": "Test",
    "location": "test",
    "phoneNumber": "1234567",
    "state": "test",
    "zip": "12345",
  },
}
`;

exports[`User testing User will be updated with Bearer token also find by id with success message 1`] = `
Object {
  "message": "User updated successfully",
  "success": true,
  "user": Object {
    "address": "update address",
    "city": "update city",
    "country": "update country",
    "email": "test@test.com",
    "firstName": "update name",
    "lastName": "update",
    "location": "update location",
    "phoneNumber": "1234567",
    "state": "update state",
    "zip": "12345",
  },
}
`;
